src/Microsoft.ML.Core/Data/Schema.cs

```diff
@@ -328,7 +328,7 @@ public ColumnType GetMetadataTypeOrNull(string kind, int col)
             var meta = this[col].Metadata;
             if (meta == null)
                 return null;
-            if (((ISchema)meta.Schema).TryGetColumnIndex(kind, out int metaCol))
+            if (meta.Schema.TryGetColumnIndex(kind, out int metaCol))
```

Comment:
It will be in another PR. My plan is to remove implementations of `ISchema` first. After discussing with @Ivanidzo4ka today, I decided to lower the priority of `ColumnBindingsBase` because of its complexity and do simple cases first (such as `FakeSchema`).